docs/topics/patterns/instant-messaging.mdx

```diff
@@ -0,0 +1,359 @@
+    for {
+      item, err := subscription.Item(ctx)
+      if err != nil {
+        panic(err)
+      }
+      switch message := item.(type) {
+      case momento.String:
+        fmt.Printf("received message as string: '%v'\n", message)
+      case momento.Bytes:
+        fmt.Printf("received message as bytes: '%v'\n", message)
+      }
+    }
+  }
+  ```
+
+  </TabItem>
+  <TabItem value="java" label="Java">
+
+  ```java
+  final TopicSubscribeResponse subscribeResponse =
+    topicClient
+      .subscribe(
+        TopicExample.CACHE_NAME,
+        TOPIC_NAME,
+        new ISubscriptionCallbacks() {
+          @Override
+          public void onItem(TopicMessage message) {
+            logger.info("Received message on topic {}: {}", TOPIC_NAME, message.toString());
+          };
+
+          @Override
+          public void onError(Throwable error) {
+            logger.error("Subscription to topic {} failed with error", TOPIC_NAME, error);
+          };
+        };).join();
+  ```
+  </TabItem>
+  <TabItem value="dotnet" label=".NET">
+
+  ```csharp
+  var subscriptionTask = Task.Run(async () =>
+  {
+    var subscribeResponse = await topicClient.SubscribeAsync(cacheName, TopicName);
+    switch (subscribeResponse)
+    {
+      case TopicSubscribeResponse.Subscription subscription:
+        try {
+          var cancellableSubscription = subscription.WithCancellation(cts.Token);
+          await foreach (var message in cancellableSubscription) {
+            switch (message)
+            {
+              case TopicMessage.Binary:
+                Logger.LogInformation("Received binary message from topic: {binaryData}", Convert.ToBase64String(binary.Value));
+                break;
+              case TopicMessage.Text text:
+                Logger.LogInformation("Received string message from topic: {message}", text.Value);
+                break;
+              case TopicMessage.Error error:
+                Logger.LogInformation("Received error message from topic: {error}", error.Message);
+                cts.Cancel();
+                break;
+            }
+          }
+        } finally {
+          subscription.Dispose();
+        }
+
+        break;
+      case TopicSubscribeResponse.Error error:
+        Logger.LogInformation("Error subscribing to a topic: {error}", error.Message);
+        cts.Cancel();
+        break;
+    }
+  });
+  ```
+  </TabItem>
+</Tabs>
+
+## Security
+
+Momento provides a [robust authorization mechanism](../develop/authentication/tokens) that limits consumers to specific resources and actions. When building an instant messaging feature, it is incredibly important to take scope into consideration so users do not intercept messages they shouldn't have access to.
+
+The pattern described above uses both cache and topic resources. Following the best practice recommended above by making our chat room the topic name, we can create policies in a couple of different ways.
```

Comment:
"making our chat room the topic name" - that reads a little confusing to me. "making our topic name match the name of our chat room"? or something like that?